src/firefly/js/charts/ui/HistogramOptions.jsx

```diff
@@ -60,6 +60,41 @@ const algorithmOptions = [  {label: 'Bayesian blocks', value: 'bayesianBlocks'},
 const binSizeOptions = [  {label: 'Number of bins:', value: 'numBins'},
     {label: 'Bin width:', value: 'binWidth'} ];
 
+
+var columnNameReducer= (colValStats, histogramParams) => {
+    if (!colValStats) {
+        return {};
+    }
+    return (inFields, action) => {
+
+        if (!inFields) {
+            return {};
+        }
+        let fieldKey = undefined;
+        if (action.type === VALUE_CHANGE) {
+            // when column name changes, update the min/max input
+            fieldKey = get(action.payload, 'fieldKey');
+            const colName = action.payload.value;
+            if (fieldKey === 'columnOrExpr') {
+
+                for (var i=0; i<colValStats.length; i++){
+                    if (colName=== colValStats[i].name){
+                        const dataMin = colValStats[i].min;
+                        const dataMax = colValStats[i].max;
+                        const binWidth = (dataMax - dataMin)/50.0;
```

Comment:
I filled width by the (max-min)/50 since the 50 is the default.   My thought is to give user a reasonable bin width to start with.  The users can change it later.   The bin width can be any valid float value.  It does not have to be bounded with (max-min)/numBins.